src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java

```diff
@@ -408,25 +408,36 @@ private String getMetaForTopic(String[] metas, int expectedLength, int curIdx, S
 
         Map<String, Map<String, String>> metaMap = new HashMap<>();
         int idx = 0;
-        for (String topic: topics) {
-            HashMap<String, String> topicMeta = new HashMap<>();
-            String meta = getMetaForTopic(topicIndexes, topics.length, idx, INDEX_CONF);
-            if (meta != null) {
-                topicMeta.put(INDEX, meta);
-            }
+        /*
+        ** to allow the use of 'topics.regex' instead of the static list of topics.
+        ** If topics.regex is specified in the config, the Connector will subscribe to all matching topics.
+        ** If topics.regex is used, mapping from topic value to Splunk metadata will not work,
+        ** so either the Headers must define the Splunk metadata, or simply rely on the HEC token
+        ** to set default index, sourcetype, etc.
+        */
 
-            meta = getMetaForTopic(topicSourcetypes, topics.length, idx, SOURCETYPE_CONF);
-            if (meta != null) {
-                topicMeta.put(SOURCETYPE, meta);
-            }
+        // If the config has no "topics" values, skip metamap formation
+        if(topics != null && topics.length != 0) {
+            for (String topic : topics) {
+                HashMap<String, String> topicMeta = new HashMap<>();
+                String meta = getMetaForTopic(topicIndexes, topics.length, idx, INDEX_CONF);
+                if (meta != null) {
+                    topicMeta.put(INDEX, meta);
+                }
 
-            meta = getMetaForTopic(topicSources, topics.length, idx, SOURCE_CONF);
-            if (meta != null) {
-                topicMeta.put(SOURCE, meta);
-            }
+                meta = getMetaForTopic(topicSourcetypes, topics.length, idx, SOURCETYPE_CONF);
```

Comment:
formatting here looks a little off, can be fixed?

Reply:
in a file, it's already in a proper format.